openpype/pipeline/creator_plugins.py

```diff
@@ -0,0 +1,180 @@
+import copy
+import collections
+
+
+class InstanceData(collections.OrderedDict):
```

Comment:
I am guessing that `instance.data["something"]` is used everywhere that using something like `attrs` or `data class` instead of dictionary is out of a question, correct?

Reply:
Well this is not pyblish instance but avalon instance :) (It's clear right?)

Reply:
Issue is that some keys can be defined and some cannot. Few of them will contain dictionary as value with another dictionary as it's value. Output can be
```
{
    "id": "pyblish.avalon.instance",
    "family": "image",
    "subset": "subset name",
    "active": true,
    "asset": "asset name",
    "task": "task name",
    "variant": "variant",
    "family_attributes": {
        "frame_start": 1001
    },
    "publish_attributes": {
        "Plugin1": {
            "plugin_attr_1": false,
            "plugin_attr_2": "black"
        }
    }
}
```
or
```
{
    "id": "pyblish.avalon.instance",
    "family": "image",
    "subset": "subset name",
    "active": true,
    "asset": "asset name",
    "task": "task name",
    "variant": "variant",
    "render_pass": "my_pass",
    "members": "layer name",
    "family_attributes": {},
    "publish_attributes": {}
}
```